### DISCUSSION TITLE
How to hide path (incl. file name)?

### DISCUSSION THREAD

**Discussion by ruestefa** [Bekaboo/dropbar.nvim#182] [Q&A] [answered]
In most windows, the file path and name either take up most of the space or cause almost all elements to be collapsed into ellipses. I also already have the file name in the tab bar and the full path in the status bar, so there's really no benefit from Dropbar showing it, too. How can I hide the file name and path?

I've tried this, but it doesn't work:

```
require('dropbar').setup{
    sources = {
        path = function(_)
            return false
        end,
    },
}
```

**Comment by Bekaboo** [comment, marked as answer]
```lua
require('dropbar').setup({
        bar = {
                sources = function(buf, _)
                        local sources = require('dropbar.sources')
                        if vim.bo[buf].ft == 'markdown' then
                                return { sources.markdown }
                        end
                        if vim.bo[buf].buftype == 'terminal' then
                                return { sources.terminal }
                        end
                        return {
                                utils.source.fallback({
                                        sources.lsp,
                                        sources.treesitter,
                                }),
                        }
                end,
        },
})
```

**Reply by ruestefa** [reply]
Thanks a lot, that works (after re-inserting the definition of `utils`)!

Note, though, that the `utils.source.fallsback({...}),` doesn't work in the master (`35988affc74c0cebb1484bcdfccfed523c0e7267`); nothing is shown. But the latest release works, so all good. (I noticed this because Packer apparently doesn't use releases but always the latest commit; time to switch to a different plugin manager...)

**Reply by mosheavni** [reply]
any chance this could be simpler? seems very hard to hide the file path...
